File: leetcode/python/509.py
class Solution:
    def fib(self, n: int) -> int:
        if n == 0 or n == 1:
            return n

        l = [0 for i in range(n + 1)]
        l[1] = 1

        for i in range(2, n + 1):
            l[i] = l[i - 1] + l[i - 2]

        return l[-1]

File: leetcode/java/509.java
class Solution {
  public int fib(int n) {
    int a = 0;
    int b = 1;

    for (int i = 1; i <= n; i++) {
      int t = a;
      a = a + b;
      b = t;
    }
    return a;
  }
}
